### ISSUE TITLE
Modeling IO Buffer Object with SVF

### ISSUE THREAD

**Issue by FrancisYL** [SVF-tools/SVF#199]
Hello,

I was wondering if it's a good idea to extend the current SVF memory model by adding an extra IO Buffer Object (NodeID = 4 with Type = Int8Ty) and Value (NodeID = 5).

std functions like `printf()`, `scanf()` do have side-effect on the IO buffer. There will be a race when multiple `printf()` trying to print to any output stream and the mod-ref results between any two `printf()` call pairs should be `ModRef`.
```C
printf("This prints first");
printf("This prints second");
```

One or more `extf_t` might be needed and external functions that perform IO operations defined in the https://github.com/SVF-tools/SVF/blob/f06eba337a17954981c6a470bcd273607c992f06/lib/Util/ExtAPI.cpp#L24 will get changed. This IO buffer modeling would benefit my project a lot and I'm glad to help with implementing that.

**Comment by yuleisui** [comment]
Yes, I am ok if you want to add the READ side-effects for the existing IO functions or others.

The following steps might be helpful:

1. add one or more side-effect enum in ExtAPI to indicate which parameter or its references will have a READ side-effect.
https://github.com/SVF-tools/SVF/blob/f06eba337a17954981c6a470bcd273607c992f06/include/Util/ExtAPI.h#L79

2. handle your case by adding load PAGEdges
https://github.com/SVF-tools/SVF/blob/f06eba337a17954981c6a470bcd273607c992f06/lib/SVF-FE/PAGBuilder.cpp#L986

3. add or modify the existing functions' side-effects in ExtAPI.cpp
https://github.com/SVF-tools/SVF/blob/f06eba337a17954981c6a470bcd273607c992f06/lib/Util/ExtAPI.cpp#L310

**Comment by yuleisui** [comment]
Once you add the READ side-effect, I am happy to have a look at your pull request. Thanks

**Comment by FrancisYL** [comment]
Thanks!
